glue/viewers/custom/qt/custom_viewer.py

```diff
@@ -752,6 +753,25 @@ def apply_roi(self, roi):
         subset_state = self._coordinator._build_subset_state(roi=roi)
         self.apply_subset_state(subset_state)
 
+    def add_data(self, data):
+        """
+        For convenience, we set x/y limits from the Matplotlib
+        dataLim box when we add data if and only if we haven't
+        set the x_min and y_min limits to something different
+        from the default 0-1 range. This way we do not
+        override custom limits set in a setup function but
+        most custom viewers will show data as expected.
+        """
+        super(CustomMatplotlibDataViewer, self).add_data(data)
+        if (self.state.x_min == 0 and self.state.x_max == 1 and self.state.y_min == 0 and self.state.y_max == 1):
+            with ignore_callback(self.state, "x_min", "x_max", "y_min", "y_max"):
+                self.state.x_min = self.axes.dataLim.xmin
+                self.state.x_max = self.axes.dataLim.xmax
+                self.state.y_min = self.axes.dataLim.ymin
+                self.state.y_max = self.axes.dataLim.ymax
+        self.limits_to_mpl()
```

Comment:
Just curious, why is this needed?

Reply:
Sorry, which line(s) are you asking about? The general idea is that without something like this the limits in the state object stay at the default values of (0,1) for custom viewers rather than getting updated when data (which is normally handled by limit helpers).

Reply:
I meant the last line - since we are updating the state from the Matplotlib viewer, why do we then need to convert the state limits back to Matplotlib? Is it in case we make any changes in the state because of aspect ratio/etc?

Reply:
In any case I'll go ahead and merge, we can always tweak later if you think this isn't needed but I don't think there's any harm in doing it.